pytket/extensions/quantinuum/backends/quantinuum.py

```diff
@@ -1511,10 +1511,39 @@ def _convert_result(
     results_selection: Optional[list[tuple[str, int]]] = None,
 ) -> BackendResult:
     if results_selection is None:
-        array_dict = {
-            creg: np.array([list(a) for a in reslist]).astype(np.uint8)
-            for creg, reslist in resultdict.items()
-        }
+        for creg in resultdict:
+            if any(["-" in res for res in resultdict[creg]]):
+                resultdict[creg] = [res.replace("-", "0") for res in resultdict[creg]]
+                warnings.warn(
+                    f"found negative value for creg: {creg}. \
+This could indicate a problem with the circuit submitted"
+                )
+        found_int_res = False
+        for x in [f"{s}" for s in range(2, 10)]:
```

Comment:
If an int result happens to consist only of 0s and 1s this will be missed?

Reply:
Yes, we need to wait for the results to contain a tag for this. See https://github.com/CQCL/pytket-quantinuum/issues/501